crates/astria-sequencer/src/transaction/mod.rs

```diff
@@ -269,9 +267,8 @@ impl ActionHandler for SignedTransaction {
                     .context("failed executing bridge sudo change")?,
             }
             state
-                .increment_transaction_index_of_action()
-                .await
-                .context("failed to increment index of action")?;
+                .increment_position_in_source_transaction()
```

Comment:
Commenting here because I can't comment on the loop above: we can express this more elegantly if we do the following changes:

1. change `put_current_source -> put_transaction_context<T: Into<TransactionContext>>(value: T) -> TransactionContext`
2. Provide `impl<'a> From<&'a SignedTransaction> for TransactionContext`
3. Add `#[derive(Copy)]` on `TransactionContext` (it only contains `Copy` types I think)
4. now you can get the initial transaction context from the signed tx `let mut transaction_context = state.put_transaction_context(self)`;
5. And instead of using `increment_position_in_source_transaction` after every iteration, use a more elegant iterator like this:

```rust
for (i, action) in (0..).zip(self.actions().iter()) {
    transaction_context.position_in_source_transaction = i;
    self.put_transaction_context(transaction_context);
}
```

Note that this relies on `impl<T> From<T> for T` being always present.